### ISSUE TITLE
NaN in forward/training — refs/pr/1 produces NaN even on a single-sentence CPU forward; main revision NaNs during GPU fine-tuning

### ISSUE THREAD

**Issue by everydaylucky2** [chandar-lab/NeoBERT#11]
Hi — running into two distinct NaN issues when using NeoBERT as a fine-tuning backbone (CLS pooling + classification heads).                                                  
                                                                                                                                                                                
Environment                                                                                                                                                                   
- torch 2.5.1+cu121, transformers 5.9.0, xformers 0.0.28.post3                                                                                                                
- GPU: NVIDIA A40 (Ampere), CPU also tested                                                                                                                                   
                                                                                                                                                                                
Issue A — main revision: NaN loss during GPU fine-tuning                                                                                                                      
Fine-tuning chandar-lab/NeoBERT (main) via AutoModel.from_pretrained(..., trust_remote_code=True) produces loss=nan from early steps, at every learning rate tried (including 5e-6), under both bf16 and fp32. A plain CPU forward of the same model returns valid (non-NaN) hidden states, so the divergence appears specific to the GPU training path (xformers attention).                                                                                                                                                         
                                                                                                                                                                                
Issue B — refs/pr/1: NaN even on a basic CPU forward                                                                                                                          
Trying the refs/pr/1 ("correct inference") revision makes it worse — a single, no-padding forward returns all-NaN: 

import torch                                                                                                                                                                  
from transformers import AutoModel, AutoTokenizer                                                                                                                             
rev = "refs/pr/1"                                                                                                                                                             
tok = AutoTokenizer.from_pretrained("chandar-lab/NeoBERT", trust_remote_code=True, revision=rev)                                                                              
m   = AutoModel.from_pretrained("chandar-lab/NeoBERT", trust_remote_code=True, revision=rev).eval()                                                                           
enc = tok(["hello world"], return_tensors="pt")                                                                                                                               
with torch.no_grad():                                                                                                                                                         
    h = m(**enc).last_hidden_state                                                                                                                                            
print(torch.isnan(h).any())   # -> True                                                                                                                                       
                                                                                                                                                                                
The model weights themselves contain no NaN (checked every parameter), so it's the forward computation producing NaN, not corrupted weights. Cleared the transformers_modules  dynamic-code cache and reproduced from a clean state.                                                                                                                         
                                                                                                                                                                                
Could you advise on the intended stable path for fine-tuning (e.g. a recommended attn_implementation, or whether xformers/FlashAttention is required vs. an eager fallback)? Happy to provide more repro detail. Thanks!
